docs-content/getting-started/fullstack-frameworks/next-js/7_advanced-config.md

```diff
@@ -33,6 +33,38 @@ all-in-one.
 
 Configure logging for your serverless cloud provider using one of our [cloud provider logging guides](https://www.highlight.io/docs/getting-started/backend-logging/hosting/overview), including [Vercel Log Drain for Highlight](https://vercel.com/integrations/highlight).
 
+## Custom Spans
+
+Follow the [Next.js Open Telemetry docs](https://nextjs.org/docs/app/building-your-application/optimizing/open-telemetry#custom-spans) to use custom span.
+
+Here's a quick example:
+
+```typescript
+// pages/api/page-router-trace.ts
+import { NextApiRequest, NextApiResponse } from 'next'
+
+import { withPageRouterHighlight } from '@/app/_utils/page-router-highlight.config'
+
+export default withPageRouterHighlight(async function handler(
+	req: NextApiRequest,
+	res: NextApiResponse,
+	highlight,
+) {
+	return new Promise<void>((resolve) => {
+		highlight?.tracer.startActiveSpan('page-router-span', async (span) => {
+			console.info('Here: /pages/api/page-router-trace.ts ⌚⌚⌚')
+
+			res.send(
+				`Trace sent! Check out this random number: ${Math.random()}`,
+			)
+			span.end()
+			resolve()
+		})
```

Comment:
update per comment below, unless you think this is the right interface for creating a span to expose to our customers